Incompatibility with scipy 1.9.0

The [experimental dependencies](https://github.com/gwpy/gwpy/actions/workflows/dependencies.yml) CI workflow has started [failing](https://github.com/gwpy/gwpy/runs/7072199168?check_suite_focus=true) because the `hanning` window was removed in SciPY 1.9.0, see https://scipy.github.io/devdocs/release.1.9.0.html#expired-deprecations:

```text
__ ERROR at setup of TestCliQtransform.test_filter_timeseries[highpass-filt0] __

The above exception was the direct cause of the following exception:
cls = <gwpy.cli.tests.test_qtransform.TestCliQtransform object at 0x7f5b696830d0>
prod = <gwpy.cli.qtransform.Qtransform object at 0x7f5b63f589d0>
    @classmethod
    @pytest.fixture
    def dataprod(cls, prod):
        cls._prod_add_data(prod)
>       prod.result = prod.get_spectrogram()
gwpy/cli/tests/test_spectrogram.py:37: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
gwpy/cli/qtransform.py:242: in get_spectrogram
    qtrans = proc_ts.q_transform(outseg=outseg, **self.qxfrm_args)
gwpy/timeseries/timeseries.py:2263: in q_transform
    data = self.whiten(asd=whiten, fduration=fduration,
gwpy/timeseries/timeseries.py:1747: in whiten
    tdw = filter_design.fir_from_transfer(1/asd.value, ntaps=ntaps,
gwpy/signal/filter_design.py:253: in fir_from_transfer
    impulse = truncate_impulse(impulse, ntaps=ntaps, window=window)
gwpy/signal/filter_design.py:207: in truncate_impulse
    window = signal.get_window(window, ntaps)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
window = 'hanning', Nx = 1024, fftbins = True
    def get_window(window, Nx, fftbins=True):
        """
        Return a window of a given length and type.
        Parameters
        ----------
        window : string, float, or tuple
            The type of window to create. See below for more details.
        Nx : int
            The number of samples in the window.
        fftbins : bool, optional
            If True (default), create a "periodic" window, ready to use with
            `ifftshift` and be multiplied by the result of an FFT (see also
            :func:`~scipy.fft.fftfreq`).
            If False, create a "symmetric" window, for use in filter design.
        Returns
        -------
        get_window : ndarray
            Returns a window of length `Nx` and type `window`
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
        >>> signal.get_window(('exponential', None, 1.), 9)
        array([ 0.011109  ,  0.03019738,  0.082085  ,  0.22313016,  0.6065[306](https://github.com/gwpy/gwpy/runs/7072199168?check_suite_focus=true#step:8:307)6,
                0.60653066,  0.22[313](https://github.com/gwpy/gwpy/runs/7072199168?check_suite_focus=true#step:8:314)016,  0.082085  ,  0.03019738])
        >>> signal.get_window(4.0, 9)
        array([ 0.08848053,  0.29[425](https://github.com/gwpy/gwpy/runs/7072199168?check_suite_focus=true#step:8:426)961,  0.56437221,  0.82160913,  0.97885093,
                0.97885093,  0.82160913,  0.56437221,  0.29425961])
        """
        sym = not fftbins
        try:
            beta = float(window)
        except (TypeError, ValueError) as e:
            args = ()
            if isinstance(window, tuple):
                winstr = window[0]
                if len(window) > 1:
                    args = window[1:]
            elif isinstance(window, str):
                if window in _needs_param:
                    raise ValueError("The '" + window + "' window needs one or "
                                     "more parameters -- pass a tuple.") from e
                else:
                    winstr = window
            else:
                raise ValueError("%s as window type is not supported." %
                                 str(type(window))) from e
            try:
                winfunc = _win_equiv[winstr]
            except KeyError as e:
>               raise ValueError("Unknown window type.") from e
E               ValueError: Unknown window type.
/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/site-packages/scipy/signal/windows/_windows.py:2234: ValueError

--- a/examples/frequencyseries/percentiles.py
+++ b/examples/frequencyseries/percentiles.py
@@ -45,7 +45,7 @@ hoft = TimeSeries.fetch_open_data('H1', 1187007040, 1187009088)
 # a number of ASDs, using the :meth:`~gwpy.timeseries.TimeSeries.spectrogram2`
 # method:
 
-sg = hoft.spectrogram2(fftlength=4, overlap=2, window='hanning') ** (1/2.)
+sg = hoft.spectrogram2(fftlength=4, overlap=2, window='hann') ** (1/2.)
 
 # From this we can trivially extract the median, 5th and 95th percentiles:
 

--- a/examples/frequencyseries/percentiles.py
+++ b/examples/frequencyseries/percentiles.py
@@ -45,7 +45,7 @@ hoft = TimeSeries.fetch_open_data('H1', 1187007040, 1187009088)
 # a number of ASDs, using the :meth:`~gwpy.timeseries.TimeSeries.spectrogram2`
 # method:
 
-sg = hoft.spectrogram2(fftlength=4, overlap=2, window='hann') ** (1/2.)
+sg = hoft.spectrogram2(fftlength=4, overlap=2, window='hann') ** (1 / 2.)
 
 # From this we can trivially extract the median, 5th and 95th percentiles:
 

--- a/gwpy/signal/filter_design.py
+++ b/gwpy/signal/filter_design.py
@@ -181,7 +181,7 @@ def truncate_transfer(transfer, ncorner=None):
     return out
 
 
-def truncate_impulse(impulse, ntaps, window='hanning'):
+def truncate_impulse(impulse, ntaps, window='hann'):
     """Smoothly truncate a time domain impulse response
 
     Parameters
@@ -193,7 +193,7 @@ def truncate_impulse(impulse, ntaps, window='hanning'):
         number of taps in the final filter
 
     window : `str`, `numpy.ndarray`, optional
-        window function to truncate with, default: ``'hanning'``
+        window function to truncate with, default: ``'hann'``
         see :func:`scipy.signal.get_window` for details on acceptable formats
 
     Returns
@@ -211,7 +211,7 @@ def truncate_impulse(impulse, ntaps, window='hanning'):
     return out
 
 
-def fir_from_transfer(transfer, ntaps, window='hanning', ncorner=None):
+def fir_from_transfer(transfer, ntaps, window='hann', ncorner=None):
     """Design a Type II FIR filter given an arbitrary transfer function
 
     Parameters
@@ -223,7 +223,7 @@ def fir_from_transfer(transfer, ntaps, window='hanning', ncorner=None):
         number of taps in the final filter, must be an even number
 
     window : `str`, `numpy.ndarray`, optional
-        window function to truncate with, default: ``'hanning'``
+        window function to truncate with, default: ``'hann'``
         see :func:`scipy.signal.get_window` for details on acceptable formats
 
     ncorner : `int`, optional

--- a/gwpy/signal/tests/test_window.py
+++ b/gwpy/signal/tests/test_window.py
@@ -29,7 +29,7 @@ __author__ = 'Duncan Macleod <duncan.macleod@ligo.org>'
 
 
 def test_canonical_name():
-    assert window.canonical_name('Hanning') == 'hann'
+    assert window.canonical_name('Han') == 'hann'
     with pytest.raises(ValueError) as exc:
         window.canonical_name('blah')
     assert str(exc.value) == ('no window function in scipy.signal '

--- a/gwpy/signal/window.py
+++ b/gwpy/signal/window.py
@@ -54,7 +54,7 @@ def canonical_name(name):
     Examples
     --------
     >>> from gwpy.signal.window import canonical_name
-    >>> canonical_name('hanning')
+    >>> canonical_name('hann')
     'hann'
     >>> canonical_name('ksr')
     'kaiser'

--- a/gwpy/timeseries/timeseries.py
+++ b/gwpy/timeseries/timeseries.py
@@ -1709,7 +1709,7 @@ class TimeSeries(TimeSeriesBase):
         return out
 
     def whiten(self, fftlength=None, overlap=0, method=DEFAULT_FFT_METHOD,
-               window='hanning', detrend='constant', asd=None,
+               window='hann', detrend='constant', asd=None,
                fduration=2, highpass=None, **kwargs):
         """Whiten this `TimeSeries` using inverse spectrum truncation
 
@@ -1728,7 +1728,7 @@ class TimeSeries(TimeSeriesBase):
 
         window : `str`, `numpy.ndarray`, optional
             window function to apply to timeseries prior to FFT,
-            default: ``'hanning'``
+            default: ``'hann'``
             see :func:`scipy.signal.get_window` for details on acceptable
             formats
 
@@ -1891,7 +1891,7 @@ class TimeSeries(TimeSeriesBase):
         # return the self-gated timeseries
         return self.mask(deadtime=deadtime, const=0, tpad=tpad)
 
-    def convolve(self, fir, window='hanning'):
+    def convolve(self, fir, window='hann'):
         """Convolve this `TimeSeries` with an FIR filter using the
            overlap-save method
 
@@ -1901,7 +1901,7 @@ class TimeSeries(TimeSeriesBase):
             the time domain filter to convolve with
 
         window : `str`, optional
-            window function to apply to boundaries, default: ``'hanning'``
+            window function to apply to boundaries, default: ``'hann'``
             see :func:`scipy.signal.get_window` for details on acceptable
             formats
 
@@ -1957,7 +1957,7 @@ class TimeSeries(TimeSeriesBase):
         out.__array_finalize__(self)
         return out
 
-    def correlate(self, mfilter, window='hanning', detrend='linear',
+    def correlate(self, mfilter, window='hann', detrend='linear',
                   whiten=False, wduration=2, highpass=None, **asd_kw):
         """Cross-correlate this `TimeSeries` with another signal
 
@@ -1968,7 +1968,7 @@ class TimeSeries(TimeSeriesBase):
 
         window : `str`, optional
             window function to apply to timeseries prior to FFT,
-            default: ``'hanning'``
+            default: ``'hann'``
             see :func:`scipy.signal.get_window` for details on acceptable
             formats
 

